src/wp-includes/template.php

```diff
@@ -965,50 +965,138 @@ function wp_finalize_template_enhancement_output_buffer( string $output, int $ph
+			sprintf(
+				/* translators: %s is the exception class name */
+				__( 'Uncaught exception "%s" thrown:' ),
+				get_class( $exception )
+			) . ' ' . $exception->getMessage(),
+			E_USER_WARNING
+		);
+		$did_just_catch_exception = false;
+	}
+
+	try {
+		/**
+		 * Fires after the template enhancement output buffer has been finalized.
+		 *
+		 * This happens immediately before the template enhancement output buffer is flushed. No output may be printed
+		 * at this action; prior to PHP 8.5, the output will be silently omitted, whereas afterward a deprecation notice
+		 * will be emitted. Nevertheless, HTTP headers may be sent, which makes this action complimentary to the
+		 * {@see 'send_headers'} action, in which headers may be sent before the template has started rendering. In
+		 * contrast, this `wp_finalized_template_enhancement_output_buffer` action is the possible point at which HTTP
+		 * headers can be sent. This action does not fire if the "template enhancement output buffer" was not started.
+		 * This output buffer is automatically started if this action is added before
+		 * {@see wp_start_template_enhancement_output_buffer()} runs at the {@see 'wp_before_include_template'} action
+		 * with priority 1000. Before this point, the output buffer will also be started automatically if there was a
+		 * {@see 'wp_template_enhancement_output_buffer'} filter added, or if the
+		 * {@see 'wp_should_output_buffer_template_for_enhancement'} filter is made to return `true`.
+		 *
+		 * Important: Because this action fires inside an output buffer callback (i.e. display handler), any callbacks
+		 * added to the action must not attempt to start their own output buffers. Otherwise, PHP will raise a fatal
+		 * error: "Cannot use output buffering in output buffering display handlers."
+		 *
+		 * @since 6.9.0
+		 *
+		 * @param string $output Finalized output buffer.
+		 */
+		do_action( 'wp_finalized_template_enhancement_output_buffer', $filtered_output );
+	} catch ( Exception $exception ) {
+		// Emit to the error log as a warning not as an error to prevent halting execution.
+		$did_just_catch_exception = true;
+		trigger_error(
+			sprintf(
+				/* translators: %s is the exception class name */
+				__( 'Uncaught exception "%s" thrown:' ),
+				get_class( $exception )
+			) . ' ' . $exception->getMessage(),
+			E_USER_WARNING
+		);
+		$did_just_catch_exception = false;
+	}
+
+	// Append any errors to be displayed before returning flushing the buffer.
+	if ( $display_errors ) {
+		foreach ( $error_log as $error ) {
+			switch ( $error['level'] ) {
+				case E_USER_NOTICE:
+					$type = 'Notice';
+					break;
+				case E_USER_DEPRECATED:
+					$type = 'Deprecated';
+					break;
+				case E_USER_WARNING:
+					$type = 'Warning';
+					break;
+				default:
+					$type = 'Error';
+			}
+			$format = "<br />\n<b>%s</b>: %s in <b>%s</b> on line <b>%d</b><br />";
+			if ( ! ini_get( 'html_errors' ) ) {
+				$format = strip_tags( $format );
```

Comment:
would we not want to use `wp_strip_all_tags()` here, or use some HTML API processor to get something as close to `innerText` as we can?

```php
echo strip_tags( 'The <em>dog</em> walked &amp; barked.' );
// The dog walked &amp; barked.
```

or is the goal to try and prevent the possibility of throwing through some user-space bug?

of note, `strip_tags()` does not parse HTML particularly well.

Reply:
Unfortunately `wp_strip_all_tags()` also does `trim()` at the end, so then it removes the initial newline. 

The goal here is just to emulate what PHP outputs when the `html_errors` setting is turned off.

I'm only looking to remove the `B` and `BR` tags from the format string. It's not operating on the underlying error message being printed.

Reply:
I've eliminated this in 2fe5542103f1a07f6c6365bd8bbaca7048aa4b9e

Reply:
this reads clearer to me, and removes the complications of misparsing and leaving character references un-decoded.